inductiva/benchmarks/methods.py

```diff
@@ -0,0 +1,348 @@
+""" 
+Methods to interact with Benchmarks.
+"""
+from collections import defaultdict
+import datetime
+import logging
+import time
+import json
+
+from typing import Any, Callable, Dict, List, Union
+from tqdm import tqdm
+
+from inductiva.resources.machines_base import BaseMachineGroup
+from inductiva.templating.manager import TemplateManager
+from inductiva.resources.machines import MachineGroup
+from inductiva.simulators.simulator import Simulator
+from inductiva.projects.project import Project
+from inductiva.resources import machine_groups
+from inductiva.tasks.task import Task
+from inductiva import users
+
+from ..localization import translator as __
+
+
+def analyze(name: str, metadata_path: str, ouput_path: str = None) -> Dict:
+    """Analyzes a benchmark.
+    This method creates a dictionary with all the relevant information about
+    the benchmark.
+    Args:
+        name (str): Name of the benchmark.
+        metadata_path (str): Path to the metadata file.
+        ouput_path (str): Path to the output file.
+            If None it will not create an output file.
+    Returns:
+        Dict: Dictionary with the information about the benchmark.
+    """
+    logging.info("Starting analysis of benchmark %s", name)
+    with Project(name, append=True) as project:
+
+        tasks = project.get_tasks()
+
+        report = {"name": name, "resources": {}}
+
+        #reads the metadata file and creates a dictionary with the task_id as
+        #key and the metadata as value
+        metadata_tasks = {}
+        with open(metadata_path, "r", encoding="utf-8") as metadata_file:
+            lines = metadata_file.readlines()
+            for line in lines:
+                temp_task = json.loads(line)
+                metadata_tasks[temp_task["task_id"]] = temp_task
+        skipped_tasks = []
+        for task in tqdm(tasks, desc="Analyzing tasks"):
+            task_status = task.get_status()
+            if task_status != "success":
+                skipped_tasks.append(task)
+                continue
+            task_info = task.info
+            mg_name = task_info.executer.vm_name.split(
+                "-")[0] + "-" + task_info.executer.vm_name.split("-")[1]
+            resource = machine_groups.get_by_name(mg_name)
```

Comment:
this is a bit cumbersome. Don't we have a direct way to get the name of the machine group?
Eitherway, no need to split the string twice:

```suggestion
            a, b, *_ = task_info.executer.vm_name.split("-")
            mg_name = a + "-" + b
            resource = machine_groups.get_by_name(mg_name)
```

Reply:
I'm getting the machine group name from the task. As far as I am aware, this is the only way to do it, since the vm_name is a combination of the machine group name and some other characters.